db/migrate/20150109083425_create_join_table_user_application_permissions.rb

```diff
@@ -0,0 +1,15 @@
+class CreateJoinTableUserApplicationPermissions < ActiveRecord::Migration
+  def change
+    create_table :user_application_permissions do |t|
+      t.integer :user_id, null: false
+      t.integer :application_id, null: false
+      t.integer :supported_permission_id, null: false
+      t.datetime :last_synced_at
+      t.timestamps
+    end
+
+    add_index :user_application_permissions, :user_id
+    add_index :user_application_permissions, [:user_id, :application_id]
+    add_index :user_application_permissions, [:user_id, :application_id, :supported_permission_id], unique: true
```

Comment:
There's no need to have both this and the previous indexes.  Queries using the `user_id` and `application_id`, or just the `user_id` will be covered by this index.  Suggest we remove the others.